sdk/communication/communication-short-codes/test/public/manageUSProgramBriefs.spec.ts

```diff
@@ -27,80 +31,155 @@ describe(`ShortCodesClient - creates, gets, updates, lists, and deletes US Progr
-      getRes = await client.getUSProgramBrief(uspb.id);
-      assertEditableFieldsAreEqual(uspb, getRes, "get after update");
+    // update program brief by calling upsert
+    if (uspb.programDetails) {
+      uspb.programDetails.callToActionUrl = updateRequest.body?.programDetails?.callToActionUrl;
+      uspb.programDetails.privacyPolicyUrl = updateRequest.body?.programDetails?.privacyPolicyUrl;
+      uspb.programDetails.termsOfServiceUrl = updateRequest.body?.programDetails?.termsOfServiceUrl;
+    }
 
+    const updateResult = await client.upsertUSProgramBrief(uspb.id, updateRequest);
+    assert.isOk(updateResult, "Update program brief failed");
+    assert.equal(uspb.id, updateResult.id, "Update program brief returned the wrong Id");
+
+    // get program brief, verify it was updated correctly
+    const getRes = await client.getUSProgramBrief(uspb.id);
+    assertEditableFieldsAreEqual(uspb, getRes, "get after update");
+  };
+
+  const _listUSProgramBriefs = async (
+    expectedProgramBriefs: USProgramBrief[],
+    byPage?: boolean,
+    itemsPerPage?: number
+  ): Promise<number> => {
+    let totalNumberOfbriefs = 0;
+    // create map of expected ids
+    const expectedBriefMap: Record<string, { brief: USProgramBrief; found: boolean }> = {};
+    expectedProgramBriefs.forEach((pb) => {
+      expectedBriefMap[pb.id] = { brief: pb, found: false };
+    });
+    if (byPage) {
+      const pages = client.listUSProgramBriefs({ top: itemsPerPage }).byPage();
+      for await (const page of pages) {
+        // loop over each item in the page
+        for (const pb of page) {
+          totalNumberOfbriefs++;
+          if (expectedBriefMap[pb.id]) {
+            expectedBriefMap[pb.id].found = true;
+            assertEditableFieldsAreEqual(
+              pb,
+              expectedBriefMap[pb.id].brief,
+              "list all program briefs, byPage"
+            );
+          }
+        }
+      }
+    } else {
       // list program briefs, validate test program brief is in the list
-      let foundTestProgramBrief = false;
       for await (const pb of client.listUSProgramBriefs()) {
-        if (pb.id === uspb.id) {
-          foundTestProgramBrief = true;
-          assertEditableFieldsAreEqual(uspb, pb, "list all program briefs");
+        totalNumberOfbriefs++;
+        if (expectedBriefMap[pb.id]) {
+          expectedBriefMap[pb.id].found = true;
+          assertEditableFieldsAreEqual(
+            pb,
+            expectedBriefMap[pb.id].brief,
+            "list all program briefs, byPage"
+          );
         }
       }
-      assert.isTrue(
-        foundTestProgramBrief,
-        "Test program brief was not returned in list of all program briefs"
-      );
+    }
+    // make sure all expected briefs were found
+    const allExpectedBriefsFound = Object.values(expectedBriefMap).every((expectedPB) => {
+      return expectedPB.found;
+    });
+    assert.isTrue(
```

Comment:
IF this assert fails, will the resulting message be helpful in finding the missing program brief?

assert.isTrue tends to not contain this information.